packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Aliases, AuthInfo } from '@salesforce/core';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+import { ForceConfigGet } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+
+export async function setupWorkspaceOrgType() {
+  const defaultUsernameOrAlias = await getDefaultUsernameOrAlias();
+  const defaultUsernameIsSet = typeof defaultUsernameOrAlias !== 'undefined';
+  let isScratchOrg = false;
+  if (defaultUsernameIsSet) {
+    const username = await getUsername(defaultUsernameOrAlias!);
+    try {
+      isScratchOrg = await isAScratchOrg(username);
+    } catch (e) {
+      if (e.name === 'NamedOrgNotFound') {
+        // If the info for a default username cannot be found,
+        // then assume that the org can be of either type
+        setDefaultUsernameHasChangeTracking(true);
+        setDefaultUsernameHasNoChangeTracking(true);
+        return;
+      } else {
+        throw e;
+      }
+    }
+  }
+  setDefaultUsernameHasChangeTracking(defaultUsernameIsSet && isScratchOrg);
+  setDefaultUsernameHasNoChangeTracking(defaultUsernameIsSet && !isScratchOrg);
+}
+
+async function isAScratchOrg(username: string): Promise<boolean> {
+  const authInfo = await AuthInfo.create(username);
+  const authInfoFields = authInfo.getFields();
+  return Promise.resolve(typeof authInfoFields.devHubUsername !== 'undefined');
+}
+
+/**
+ * Returns the non-aliased username
+ * @param usernameOrAlias
+ */
+async function getUsername(usernameOrAlias: string): Promise<string> {
+  const username = await Aliases.fetch(usernameOrAlias);
+  if (username) {
+    return Promise.resolve(username);
+  }
+  return Promise.resolve(usernameOrAlias);
+}
+
+function setDefaultUsernameHasChangeTracking(val: boolean) {
```

Comment:
Renamed these methods and their corresponding context variables to `setDefaultUsernameHasChangeTracking` and `setDefaultUsernameHasNoChangeTracking`.

Reply:
Not with the current way that I'm using it. `sfdx:default_username_has_change_tracking` is `true` only if 2 conditions are satisfied: 1) a default username is set, and 2) the org that the username is set to has change tracking. It'll return `false` if no default username is set, or the default username is set and the org it is set to doesn't having change tracking. 

Because of this, I can't use `!sfdx:default_username_has_change_tracking` to expose the change-set-based commands because that expression could return `true` if no default username is set

Reply:
And this is because we want to show both change set and non change set commands in certain situations, right ?

Reply:
Yep, there are times when we want to show both and times when we want to hide both too.